tests/objects.test.ts

```diff
@@ -0,0 +1,28 @@
+import { isMatching, P } from '../src';
+import { Equal, Expect } from '../src/types/helpers';
+
+describe('Objects', () => {
+  it('should work with symbols', () => {
+    const symbolA = Symbol('symbol-a');
+    const symbolB = Symbol('symbol-b');
+    const obj: { [symbolA]: { [symbolB]: 'foo' | 'bar' } } = {
+      [symbolA]: { [symbolB]: 'foo' },
+    };
+    if (isMatching({ [symbolA]: { [symbolB]: 'foo' } }, obj)) {
+      type t = Expect<Equal<typeof obj, { [symbolA]: { [symbolB]: 'foo' } }>>;
+    } else {
+      throw new Error('Expected obj to match the foo pattern!');
+    }
```

Comment:
This test only wasn't enough as it was passing before (as it was stripping all symbols)

So I added the other `if` for the negative check